src/main/java/com/exadel/movie_new/service/MovieServiceImpl.java

```diff
@@ -0,0 +1,86 @@
+package com.exadel.movie_new.service;
+
+import com.exadel.movie_new.model.Movie;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import org.springframework.stereotype.Service;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class MovieServiceImpl implements MovieService {
+
+
+    @Override
+    public Movie addMovie(Movie movie) {
+        List<Movie> movieData = readJsonData();
+        Movie moviePresent = null;
+
+        if (movieData != null) {
+            moviePresent = movieData.parallelStream().filter(mov -> mov.getId().equals(movie.getId())).findAny().orElse(null);
+
+        } else {
+            movieData = new ArrayList<>();
+        }
+        if (moviePresent == null) {
+            movieData.add(movie);
+            writeJsonData(movieData);
+        }
+        return movie;
+    }
+
+
+    @Override
+    public Movie getMovieDetails(String movieId) {
+        List<Movie> movieData = readJsonData();
+        return movieData.parallelStream().filter(mov -> mov.getId().equals(movieId)).findAny().orElse(null);

```

Comment:
null should not be as a response. It's better to throw custom exception like "MovieHasNotBeenFoundException". Then you must provide **exceptionHandler** , catch all custom exceptions there, and prepare response for client, something like "Sorry but film with ID {movieID} doesn't found". But this is comment for future pull request.